multi-cluster-gateway/multi-cluster-gateway-internal-bluegreen/README.md

```diff
@@ -0,0 +1,239 @@
+# Multi-Cluster Gateway Blue/Green Cluster Pattern (internal)
+
+Multi-cluster Gateway for GKE is a cloud-hosted [Gateway API](https://gateway-api.sigs.k8s.io/) controller for GKE clusters. It's a Google-hosted service that supports deploying shared load balancing resources across clusters and across regions.
+
+> Note: in this guide, *MCG* will be used as shorthand for Multi-cluster Gateway.
+
+### Use-case
+
+The MCG Blue/Green cluster pattern is designed to address Kubernetes cluster lifecycle use cases where a given GCP region has two or more GKE clusters hosting the same application(s). 
+
+Redundant GKE clusters are commonly deployed so that one cluster can be removed from service at a time, upgraded, and returned to service, while the other cluster(s) continue to service client traffic. Within this example recipe, we will leverage the Gateway API's expressive feature surface to implement the following:
+
+- Traffic mirroring to a new set of resources to confirm readiness for release
+- Traffic weighting across clusters to enable the [canary release](https://martinfowler.com/bliki/CanaryRelease.html) pattern
+- Header-based routing to target specific clusters
+
+In this example, both clusters reside in the same GCP region to demonstrate a blue/green upgrade pattern where one cluster at a time can be removed from service, upgraded, and returned to service, all while clients can continue to access a given application. Additionally, this example uses a `GatewayClass` of type `gke-l7-rilb-mc`, meaning the MCG endpoint will be a *private* IP address within your VPC. For testing, the infrastructure build instructions include the deployment of a GCE virtual machine in the same VPC that will be used as the test client.
+
+### Relevant documentation
+
+> TODO: add links to public docs
```

Comment:
You want to do this before this PR is merged ?

Reply:
i'm good with leaving it as-is, but waiting to hear from @mark-church